README.adoc

```diff
@@ -20,16 +23,51 @@ See the official documentation on https://docs.teragrep.com[docs.teragrep.com].
 == Limitations
 
 // If your project has limitations, please list them. Otherwise remove this section.
+All planned configuration sources are not yet supported.
 
-== How to [compile/use/implement]
+== How to use
 
 // add instructions how people can start to use your project
+=== Configuration
+All sources of configuration can be used with the `Configuration` interface. It provides the method `asMap()` which returns the immutable configurations.
+
+An example with `PathConfiguration`:
+[,java]
+----
+Configuration configuration = new PathConfiguration("file/path");
+Map<String, String> configurationMap = configuration.asMap();
+----
+
+=== Default configuration
+You might want to specify default configurations in case something brakes. For example, the file path in the code block above might not be found. In the following example, notice how the Map with default configurations are given through the `ImmutabilitySupportedMap` object to ensure type safety. `DefaultConfiguration` only takes an `ImmutableMap` (provided in CNF-01) as a parameter for the defaults.
+[,java]
+----
+Map<String, String> map = new HashMap<>();
+ImmutableMap<String, String> defaults = new ImmutabilitySupportedMap<>(map).toImmutableMap();
+
+DefaultConfiguration defaultConfiguration = new DefaultConfiguration(
+        new PathConfiguration("file/path"),
+        defaults
+);
+Map<String, String> configurationMap = defaultConfiguration.asMap();
+----
+
+=== Configuration objects in your project
+The configuration Map from CNF-01 shouldn't be used directly in regular objects. It should only be passed to objects that are responsible for providing configured versions of other objects. In other words, Factories.
```

Comment:
"...for providing configured versions of other objects. In other words, Factories."

The last sentence is missing out a verb. Please combine it with the previous one or modify it into a proper sentence.